GetMessage/userPreferenceChecker/__tests__/messageReadStatusAuth.test.ts

```diff
@@ -0,0 +1,261 @@
+import * as TE from "fp-ts/TaskEither";
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+
+import { Semver } from "@pagopa/ts-commons/lib/strings";
+
+import { CosmosErrors } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model";
+import { ProfileModel } from "@pagopa/io-functions-commons/dist/src/models/profile";
+import {
+  AccessReadMessageStatusEnum,
+  RetrievedServicePreference,
+  ServicesPreferencesModel
+} from "@pagopa/io-functions-commons/dist/src/models/service_preference";
+
+import { canAccessMessageReadStatus } from "../messageReadStatusAuth";
+
+import {
+  aFiscalCode,
+  aRetrievedProfile,
+  aServiceId,
+  autoProfileServicePreferencesSettings,
+  legacyProfileServicePreferencesSettings
+} from "../../../__mocks__/mocks";
+
+const MIN_READ_STATUS_PREFERENCES_VERSION = "1.5.3" as Semver;
+const PREV_APP_VERIONS = "1.3.8" as Semver;
+const NEWER_APP_VERSION = "3.1.1" as Semver;
+
+const aRetrievedServicePreference: RetrievedServicePreference = {
+  fiscalCode: aFiscalCode,
+  serviceId: aServiceId,
+  isInboxEnabled: true,
+  isEmailEnabled: true,
+  isWebhookEnabled: true
+} as RetrievedServicePreference;
+
+// --------------------
+
+const mockProfileFindLastVersionByModelId = jest.fn();
+const mockServicePreferencesFind = jest.fn();
+
+const profileModel = ({
+  //TaskEither<CosmosErrors, O.Option<TR>>
+  findLastVersionByModelId: mockProfileFindLastVersionByModelId
+} as unknown) as ProfileModel;
+
+const servicePreferenceModel = ({
+  find: mockServicePreferencesFind
+} as unknown) as ServicesPreferencesModel;
+
+// ----------------------------
+// Tests
+// ----------------------------
+
+describe("canAccessMessageReadStatus |> ok", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should return false if profile version is UNKNOWN", async () => {
+    mockProfileFindLastVersionByModelId.mockReturnValueOnce(
+      TE.of(O.some(aRetrievedProfile))
+    );
+
+    const res = await canAccessMessageReadStatus(
+      profileModel,
+      servicePreferenceModel,
+      MIN_READ_STATUS_PREFERENCES_VERSION
+    )(aServiceId, aFiscalCode)();
+
+    expect(res).toStrictEqual(E.right(false));
+
+    // Do not call service preferences if app version is UNKNOWN
+    expect(mockServicePreferencesFind).not.toHaveBeenCalled();
+  });
+
+  it("should return false if profile version is < MIN_READ_STATUS_PREFERENCES_VERSION", async () => {
```

Comment:
There are many version on profile (version, settingsVersion, lastAppVersion) we should try to reduce misunderstanding.
```suggestion
  it("should return false if profile lastAppVersion is < MIN_READ_STATUS_PREFERENCES_VERSION", async () => {
```

Reply:
Thanks, done in [b0e499c](https://github.com/pagopa/io-functions-services/pull/211/commits/b0e499c653e80b6bd6921f9b84d803d0e26f9bfd)